cmd/s3-gw/app_healthy.go

```diff
@@ -1,46 +1,46 @@
 package main
 
 import (
-	"fmt"
-	"net/http"
+	"github.com/prometheus/client_golang/prometheus"
+)
 
-	"github.com/gorilla/mux"
+const (
+	namespace      = "neofs_s3_gw"
+	stateSubsystem = "state"
 )
 
-// Healthy is a health check interface.
-type Healthy interface {
-	Status() error
+type GateMetrics struct {
+	stateMetrics
 }
 
-const (
-	healthyState       = "NeoFS S3 Gateway is "
-	hdrContentType     = "Content-Type"
-	defaultContentType = "text/plain; charset=utf-8"
-)
+type stateMetrics struct {
+	healthCheck prometheus.Gauge
+}
+
+func newGateMetrics() *GateMetrics {
+	stateMetric := newStateMetrics()
+	stateMetric.register()
+
+	return &GateMetrics{
+		stateMetrics: stateMetric,
+	}
+}
+
+func newStateMetrics() stateMetrics {
```

Comment:
I’m not sure but function starting with “new” and not returning a pointer — doesn’t it seem confusing?

Reply:
Maybe we can drop the function? And move the content into ‘newGateMetrics’?